executor/config/device_config.py

```
@@ -0,0 +1,381 @@
+    Returns:
+        DeviceConfig instance
+
+    Raises:
+        FileNotFoundError: If specified config_path doesn't exist
+        json.JSONDecodeError: If config file contains invalid JSON
+    """
+    # Determine config file path
+    if config_path:
+        path = Path(config_path)
+        if not path.exists():
+            raise FileNotFoundError(f"Config file not found: {config_path}")
```

Ruff (0.15.0)

[warning] Avoid specifying long messages outside the exception class

(TRY003)